src/components/CR8 - GalleryText/GalleryText.vue

```diff
@@ -0,0 +1,247 @@
+                    <component 
+                        :is="`logo-${company.name}`"
+                        class="company-logo"
+                        @mouseover="set(company)" 
+                        @mouseout="set(null)" />
+                </nuxt-link>
+                <span class="text">{{company.text}} </span>
+            </span>
+        </div>
+
+        <!-- Images absolute -->
+        <transition
+            v-for="(image, index) in images"
+            :key="image.id"
+            name="fade"
+            mode="out-in"
+        >
+            <wp-image
+                v-show="index == activeImage"
+                class="image"
+                :image="image"
+                mode="fullbleed"
+            />
+        </transition>
+
+        <!-- Bottom Text -->
+        <p class="end-text">These diverse talent are connected by a creative singularity.</p>
+        <p class="end-text">Together we are <logo-makemake class="company-logo is-makemake" /></p>
+
+    </div>
+</template>
+
+<script>
+import WpImage from "@/components/global/WpImage"
+import NuxtLink from "@/components/global/NuxtLink"
+import SvgArrowRight from "@/assets/svgs/arrow-right.svg"
+
+// Company Logos
+import LogoRockPaperScissors from '@/assets/svgs/companies/logo-rock-paper-scissors.svg'
+import LogoElastic from '@/assets/svgs/companies/logo-elastic.svg'
+import LogoA52 from '@/assets/svgs/companies/logo-a52.svg'
+import LogoPrimary from '@/assets/svgs/companies/logo-primary.svg'
+import LogoJax from '@/assets/svgs/companies/logo-jax.svg'
+import LogoIndestructible from '@/assets/svgs/companies/logo-indestructible.svg'
+
+import LogoMakemake from '@/assets/svgs/companies/logo-makemake.svg'
+
+export default {
+    components: { 
+        WpImage, 
+        NuxtLink, 
+        SvgArrowRight, 
+        //Company Logos
+        LogoRockPaperScissors, 
+        LogoElastic, 
+        LogoA52, 
+        LogoPrimary, 
+        LogoJax, 
+        LogoIndestructible, 
+        LogoMakemake
+    },
+    props: {
+        companies: {
+            type: Array,
+            default: () => []
+        }
+    },
+    data () {
+        return {
+            company: null,
+            timer: null,
+            activeImage: 0
+        }
+    },
+    computed: {
+        images() {
+            return this.company ? this.company.images : []
+        },
+        imagesLength () {
+            return this.images.length
+        }
+    },
+    methods: {
+        set(company) {
+            if (!company) return this.reset()
+            this.company = company
+            this.start()
+        },
+        start() {
+            if (!this.timer) {
+                this.timer = setInterval( () => {               
+                    if (this.activeImage < this.imagesLength) {
+                        this.activeImage++
+                    } else {
+                        this.activeImage = 0
+                    }
+                }, 800)
+            }
```

Comment:
Can you explain why you are doing it this way, and not how I noted in a previous note?

If I follow this logic correctly, you are not resetting the interval on on each new hover, so it's possible to come into a hover while it's already half way through the interval.

I think tit should be this:

```
        start() {
                clearInterval(this.timer)
                this.timer = setInterval( () => {               
                    if (this.activeImage < this.imagesLength) {
                        this.activeImage++
                    } else {
                        this.activeImage = 0
                    }
                }, 800)
            }
     }
```

That should restart the interval timer on each new hover.

Reply:
i actually missed this.  Sorry about that.  Only saw the if else statement and assumed you didn't want the ternary op.  did not see the reset.  I see what you are doing.  I had the @mouseout reset the interval but i can see why you would want to make sure the interval is reset before a new one is started just in case.  Also avoids having to nest the if else inside another if statement.